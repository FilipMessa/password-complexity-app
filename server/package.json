{
  "name": "server",
  "version": "1.0.0",
  "main": "dist/index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/cors": "^2.8.6",
    "@types/express": "^4.17.2",
    "@types/helmet": "^4.0.0",
    "@types/morgan": "^1.7.37",
    "@types/node": "^15.0.2",
    "@types/supertest": "^2.0.8",
    "@types/zxcvbn": "^4.4.0",
    "supertest": "^6.1.3",
    "ts-node-dev": "^1.0.0-pre.44"
  },
  "dependencies": {
    "body-parser": "^1.19.0",
    "cors": "^2.8.5",
    "express": "^4.17.3",
    "helmet": "^4.6.0",
    "morgan": "^1.9.1",
    "password-strength": "1.0.0",
    "winston": "^3.2.1",
    "zxcvbn": "^4.4.2"
  },
  "scripts": {
    "build": "tsc -p .",
    "start": "node dist/app.js",
    "dev": "ts-node-dev --respawn  src/index.ts",
    "test": "jest --passWithNoTests"
  }
}
