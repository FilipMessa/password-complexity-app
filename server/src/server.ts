import express, { NextFunction, Request, Response } from 'express';
import { handleError, notFound, setGlobalMiddlewares } from './middlewares';
import { AppError } from './modules/errors';
import passwordComplexity from './routes/passwordComplexity';

export const server = express();

setGlobalMiddlewares(server);

server.use('/password-complexity', passwordComplexity);

server.use(notFound);
server.use((err: AppError, _req: Request, res: Response, next: NextFunction) => {
  handleError(err, res);
  next();
});
