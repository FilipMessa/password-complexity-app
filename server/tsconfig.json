{
  "extends": "../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "dist"
  }
  //"lib": ["es2015"]
}
