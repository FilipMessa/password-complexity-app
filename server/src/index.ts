import { server } from './server';

const PORT = process.env.PORT || 3005;
server.listen(PORT, () => {
  return console.log(`server is listening on ${PORT}`);
});
