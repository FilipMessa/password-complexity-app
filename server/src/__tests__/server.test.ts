import supertest from 'supertest';
import { server } from '../server';
const request = supertest(server);

test('should return not found error', async () => {
  const res = await request.post('/unknown-api').send({
    password: 'aaaaa',
  });
  expect(res.body).toMatchSnapshot();
});

['v1', 'v2'].forEach((apiVersion) => {
  describe(`/password-complexity/${apiVersion}`, () => {
    it('should return socpre 0', async () => {
      const res = await request.post('/password-complexity/v2').send({
        password: 'aaaaa',
      });
      expect(res.body).toMatchSnapshot();
    });

    it('should return socpre 1', async () => {
      const res = await request.post(`/password-complexity/${apiVersion}`).send({
        password: 'aaaaa1',
      });
      expect(res.body).toMatchSnapshot();
    });

    it('should return socpre 2', async () => {
      const res = await request.post(`/password-complexity/${apiVersion}`).send({
        password: '1aaa!',
      });
      expect(res.body).toMatchSnapshot();
    });

    it('should return socpre 3', async () => {
      const res = await request.post(`/password-complexity/${apiVersion}`).send({
        password: '1aaa!A',
      });
      expect(res.body).toMatchSnapshot();
    });

    it('should return socpre 4', async () => {
      const res = await request.post(`/password-complexity/${apiVersion}`).send({
        password: '1aaa!A@@2!',
      });
      expect(res.body).toMatchSnapshot();
    });

    it('should return bad request error', async () => {
      const res = await request.post(`/password-complexity/${apiVersion}`).send({
        psw: '1235667a',
      });

      expect(res.body).toMatchSnapshot();
    });
  });
});
