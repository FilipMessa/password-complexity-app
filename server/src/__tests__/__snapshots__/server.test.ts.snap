// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/password-complexity/v1 should return bad request error 1`] = `
Object {
  "message": "The password property is missing in Post body or have invalid value",
  "status": "error",
  "statusCode": 400,
  "type": "Bad Request",
}
`;

exports[`/password-complexity/v1 should return socpre 0 1`] = `
Object {
  "score": 0,
  "statusCode": 200,
  "strength": "Worst",
}
`;

exports[`/password-complexity/v1 should return socpre 1 1`] = `
Object {
  "score": 1,
  "statusCode": 200,
  "strength": "Bad",
}
`;

exports[`/password-complexity/v1 should return socpre 2 1`] = `
Object {
  "score": 1,
  "statusCode": 200,
  "strength": "Bad",
}
`;

exports[`/password-complexity/v1 should return socpre 3 1`] = `
Object {
  "score": 1,
  "statusCode": 200,
  "strength": "Bad",
}
`;

exports[`/password-complexity/v1 should return socpre 4 1`] = `
Object {
  "score": 3,
  "statusCode": 200,
  "strength": "Good",
}
`;

exports[`/password-complexity/v2 should return bad request error 1`] = `
Object {
  "message": "The password property is missing in Post body or have invalid value",
  "status": "error",
  "statusCode": 400,
  "type": "Bad Request",
}
`;

exports[`/password-complexity/v2 should return socpre 0 1`] = `
Object {
  "score": 0,
  "statusCode": 200,
  "strength": "Worst",
}
`;

exports[`/password-complexity/v2 should return socpre 1 1`] = `
Object {
  "score": 1,
  "statusCode": 200,
  "strength": "Bad",
}
`;

exports[`/password-complexity/v2 should return socpre 2 1`] = `
Object {
  "score": 2,
  "statusCode": 200,
  "strength": "Weak",
}
`;

exports[`/password-complexity/v2 should return socpre 3 1`] = `
Object {
  "score": 3,
  "statusCode": 200,
  "strength": "Good",
}
`;

exports[`/password-complexity/v2 should return socpre 4 1`] = `
Object {
  "score": 4,
  "statusCode": 200,
  "strength": "Strong",
}
`;

exports[`should return not found error 1`] = `
Object {
  "message": "Requested resources was not found.",
  "status": "error",
  "statusCode": 404,
  "type": "Not Found",
}
`;
